dynawo/sources/Models/CPP/ModelNetwork/DYNModelLine.cpp

```diff
@@ -972,16 +974,18 @@ ModelLine::evalZ(const double& t) {
                 throw DYNError(Error::MODELER, DynamicLineStatusNotSupported, id_);
               switch (getConnectionState()) {
               case OPEN:
-                network_->addEvent(id_, DYNTimeline(LineCloseSide2));
+                if (network_->hasTimeline()) network_->addEvent(id_, DYNTimeline(LineCloseSide2));
                 modelBus2_->getVoltageLevel()->connectNode(modelBus2_->getBusIndex());
                 break;
               case CLOSED:
-                network_->addEvent(id_, DYNTimeline(LineOpenSide1));
+                if (network_->hasTimeline()) network_->addEvent(id_, DYNTimeline(LineOpenSide1));
```

Comment:
It is more readable to put this in 2 lines

Reply:
Yes, but it depends upon who reads the line. As rules are not strictly defined, I won't discuss the coding style. I prefer to ask clang-format to do the job in an objective way than discussing about how to write code. In that particular case it was easier to write since that condition appears in other places

Reply:
anyway I would prefer a solution in which this test is included in the addEvent method.

Reply:
Florentine, it would have been more clear and a lot simpler. I would have done it but you asked not to call addEvent() when it was not usefull. In this way we don't call addEvent() when it is not needed, saving some little time, but code is more complex. In the addEvent() I add an assert() that tells not to call addEvent().

Reply:
I think there is a misunderstanding: I asked to not build the TimelineMessage when not needed, not to avoid calling addEvent. You can also change the addEvent parameters. Please update this and also the associated Modelica model equivalent methods as today this development cannot working.

Reply:
Also, please amend the commit so that its description explains what it does

Reply:
commit is amended in github

Reply:
Also, I don't get you. addEvent() is the way to build a Timeline message. I avoid calling it when not needed. Sometimes addEvent() has a parameter of type MessageLine that is build inside. avoiding addEvent() will avoid calling MessageLine(). In the case I let MessageLine() build a MessageLine, it is something I did not see. I make a review about it.

Reply:
Let me rephrase: I would like a solution in which the hasTimeline test is inside the addEvent method and we do not build the instance of TimelineMessage if !hasTimeline

Reply:
regarding the commit message, adding a message on github is not equivalent. Please use the command 
git commit --amend
and then it will be updated in this PR at the next push you will make

Reply:
OK; regarding what we said, I re-wrote a lot, almost all the code I pushed. It will be easier to delete the branch s I can start a brand new one. that on was WIP and not supposed to be merged; a new branch will please everybody. That is why this push is on a date, like a draft, not supposed to be kept. Now it is quite hard for me to add something in the branch; I tried this morning and received an all bunch of injuries from git :(

Reply:
git push --force-with-lease

Reply:
it works now